addons/mail/static/src/js/form_renderer.js

```diff
@@ -1,77 +1,151 @@
+               await this._chatterComponent.mount(this._chatterEl, { position: 'self' });
+            */
+            await this._chatterComponent.mount(this._chatterEl);
+        }
+    },
+    /**
+     * Delete the chatter component
      *
+     * @private
+     */
+    _deleteChatter() {
+        if (this._chatterComponent) {
+            this._chatterComponent.destroy();
+            this._chatterComponent = undefined;
+        }
+        this.env.store.dispatch('deleteChatter', this._chatterLocalId);
+    },
+    /**
+     * Force mount the chatter
+     *
+     * FIXME {XDU} It's a hack : chatterComponent is in a fragment :
+     * - no more in the DOM
+     * - no more correctly connected to the store
+     * We need to retarget it correctly in the DOM to fix that issue
+     * The issue is that the component considers itself as mounted
+     * but is no more added to the view as the super call to render view
+     * re-rendered the view entirely and without this component.
+     * @private
+     */
+    async _forceMountChatterComponent() {
+        this._chatterComponent.__owl__.isMounted = false;
+        /*  FIXME {xdu}
+            could be better to mount in "replace" mode but the mount is failing with that mode
+            await this._chatterComponent.mount(this._chatterEl, { position: 'self' });
+         */
+        if (this._chatterEl) {
+            await this._chatterComponent.mount(this._chatterEl);
+        }
+    },
+   /**
      * @override
      * @private
      */
-    _renderNode: function (node) {
-        var self = this;
-        if (node.tag === 'div' && node.attrs.class === 'oe_chatter') {
-            if (!this.chatter) {
-                this.chatter = new Chatter(this, this.state, this.mailFields, {
-                    isEditable: this.activeActions.edit,
-                    viewType: 'form',
-                });
+   _renderNode(node) {
+       if (node.tag === 'div' && node.attrs.class === 'oe_chatter') {
+           this._hasChatter = true;
+           const $el = $('<div class="o_Chatter_container"/>');
+           this._chatterEl = $el[0];
+           return $el;
+       }
+       else {
+           return this._super(...arguments);
+       }
+   },
+    /**
+     * Overrides the function to render the chatter once the form view is rendered.
+     *
+     * @override
+     * @private
+     */
+    async _renderView() {
+        await this._super(...arguments);
 
-                var $temporaryParentDiv = $('<div>');
-                this.defs.push(this.chatter.appendTo($temporaryParentDiv).then(function () {
-                    self.chatter.$el.unwrap();
-                    self._handleAttributes(self.chatter.$el, node);
-                }));
-                return $temporaryParentDiv;
+        if (this._hasChatter && !this._preventChatterLoad) {
+            Chatter.env = this.env;
+            if (!this._chatterComponent) {
+                await this._createChatter();
```

Comment:
You mean just create the chatter here and then wait `on_attach_callback` to do the mount ?

Reply:
By the way, shouldn't we wait to see if JS team managed to make an universal adapter ? Just to avoid to change for the Xth time the mounting process (which will be changed again if the adapter comes out) ?

Reply:
I think it should mount chatter here only if in DOM, otherwise `on_attach_callback` should do it at a later date.

TBH, I wouldn't wait for their adapter, even though it looks a bit interesting if it works.

Reply:
Ok I See, I just added a`this.isInDom` in `createChatter` function before mounting the component

Reply:
maybe would be clearer if mounting of chatter is moved away from `_createChatter()`